userland/src/add_two_ints_client.cpp

```diff
@@ -0,0 +1,43 @@
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <sys/time.h>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <simple_msgs/AllBuiltinTypes.h>
+#include <simple_msgs/AllDynamicArrayTypes.h>
+#include <simple_msgs/AllPrimitiveTypes.h>
+#include <simple_msgs/AllStaticArrayTypes.h>
+#include <simple_msgs/Nested.h>
+#include <simple_msgs/String.h>
+#include <simple_msgs/Uint32.h>
+
+#include <userland_msgs/AddTwoInts.h>
+
+int main(int argc, char** argv)
+{
+  rclcpp::init(argc, argv);
+
+  auto node = rclcpp::Node::make_shared("add_two_ints_client");
+
+  auto client = node->create_client<userland_msgs::AddTwoInts>("add_two_ints");
```

Comment:
I was expecting something like this for the asynchronous case:

```c++
void handle_response(const std::shared_ptr<userland_msgs::AddTwoInts::Response> &response)
{
  std::cout << "Got response: " << response->sum << std::endl;
}

...
client->async_send_request(request, handle_response);  // I guess you pass a pre-allocated response too
rclcpp::spin();  // This could be here or in another thread and would cause the callback to get called
```